x/reward/client/cli/cli_test.go

```diff
@@ -0,0 +1,1023 @@
+			[]string{
+				"active",
+			},
+			false,
+			false,
+			"",
+			0,
+			[]uint64{1},
+		},
+		{"query pending reward programs",
+			[]string{
+				"pending",
+			},
+			false,
+			false,
+			"",
+			0,
+			[]uint64{3, 5},
+		},
+		{"query completed reward programs",
+			[]string{
+				"completed",
+			},
+			false,
+			false,
+			"",
+			0,
+			[]uint64{2, 4},
+		},
+		{"query outstanding reward programs",
+			[]string{
+				"outstanding",
+			},
+			false,
+			false,
+			"",
+			0,
+			[]uint64{1, 3, 5},
+		},
+		{"query by id reward programs",
+			[]string{
+				"2",
+			},
+			true,
+			false,
+			"",
+			0,
+			[]uint64{2},
+		},
+		{"query by id reward programs",
+			[]string{
+				"99",
+			},
+			true,
+			true,
+			"failed to query reward program 99: rpc error: code = Unknown desc = rpc error: code = Internal desc = unable to query for reward program by ID: reward program not found: unknown request",
+			0,
+			[]uint64{2},
+		},
+		{"query invalid query type",
+			[]string{
+				"invalid",
+			},
+			true,
+			true,
+			"invalid argument arg : invalid",
+			0,
+			[]uint64{},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			clientCtx := s.network.Validators[0].ClientCtx
+			out, err := clitestutil.ExecTestCLICmd(clientCtx, rewardcli.GetRewardProgramCmd(), append(tc.args, []string{fmt.Sprintf("--%s=json", tmcli.OutputFlag)}...))
+			if tc.expectErr {
+				s.Assert().Error(err)
+				s.Assert().Equal(tc.expectErrMsg, err.Error())
+			} else if tc.byId {
+				var response types.QueryRewardProgramByIDResponse
+				s.Assert().NoError(err)
+				err = s.cfg.Codec.UnmarshalJSON(out.Bytes(), &response)
+				s.Assert().NoError(err)
+				s.Assert().Equal(tc.expectedIds[0], response.RewardProgram.Id)
+			} else {
+				var response types.QueryRewardProgramsResponse
+				s.Assert().NoError(err)
+				err = s.cfg.Codec.UnmarshalJSON(out.Bytes(), &response)
+				s.Assert().NoError(err)
+				s.Assert().Equal(len(tc.expectedIds), len(response.RewardPrograms))
+				//s.Assert().Equal(tc.expectedOutput, response)
+				for _, expectedId := range tc.expectedIds {
+					s.Assert().True(containsRewardProgramId(response.RewardPrograms, expectedId))
+				}
+			}
+		})
+	}
+}
+
+func containsRewardProgramId(rewardPrograms []types.RewardProgram, id uint64) bool {
+	for _, rewardProgram := range rewardPrograms {
+		if rewardProgram.Id == id {
+			return true
+		}
+	}
+	return false
+}
+
+func containsClaimPeriodId(claimPeriodDist []types.ClaimPeriodRewardDistribution, id uint64) bool {
+	for _, dist := range claimPeriodDist {
+		if dist.ClaimPeriodId == id {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *IntegrationTestSuite) TestQueryClaimPeriodRewardDistributionAll() {
+	defaultMaxQueryIds := make([]uint64, 100)
+	for i := 0; i < 100; i++ {
+		defaultMaxQueryIds[i] = uint64(i + 1)
+	}
+	testCases := []struct {
+		name         string
+		args         []string
+		byId         bool
+		expectErr    bool
```

Comment:
Yep good point.  I will keep my eye out for this pattern else where in our project.  Since, we copied and pasted the structure from some other module.